sdk/trafficmanager/Azure.ResourceManager.TrafficManager/src/Generated/UserMetricsModelResource.cs

```diff
@@ -0,0 +1,189 @@
+        internal UserMetricsModelResource(ArmClient client, ResourceIdentifier id) : base(client, id)
+        {
+            _userMetricsModelTrafficManagerUserMetricsKeysClientDiagnostics = new ClientDiagnostics("Azure.ResourceManager.TrafficManager", ResourceType.Namespace, Diagnostics);
+            TryGetApiVersion(ResourceType, out string userMetricsModelTrafficManagerUserMetricsKeysApiVersion);
+            _userMetricsModelTrafficManagerUserMetricsKeysRestClient = new TrafficManagerUserMetricsKeysRestOperations(Pipeline, Diagnostics.ApplicationId, Endpoint, userMetricsModelTrafficManagerUserMetricsKeysApiVersion);
+#if DEBUG
+			ValidateResourceId(Id);
+#endif
+        }
+
+        /// <summary> Gets the resource type for the operations. </summary>
+        public static readonly ResourceType ResourceType = "Microsoft.Network/trafficManagerUserMetricsKeys";
+
+        /// <summary> Gets whether or not the current instance has data. </summary>
+        public virtual bool HasData { get; }
+
+        /// <summary> Gets the data representing this Feature. </summary>
+        /// <exception cref="InvalidOperationException"> Throws if there is no data loaded in the current instance. </exception>
+        public virtual UserMetricsModelData Data
+        {
+            get
+            {
+                if (!HasData)
+                    throw new InvalidOperationException("The current instance does not have data, you must call Get first.");
+                return _data;
+            }
+        }
+
+        internal static void ValidateResourceId(ResourceIdentifier id)
+        {
+            if (id.ResourceType != ResourceType)
+                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, "Invalid resource type {0} expected {1}", id.ResourceType, ResourceType), nameof(id));
+        }
+
+        /// <summary>
+        /// Delete a subscription-level key used for Real User Metrics collection.
+        /// Request Path: /subscriptions/{subscriptionId}/providers/Microsoft.Network/trafficManagerUserMetricsKeys/default
+        /// Operation Id: TrafficManagerUserMetricsKeys_Delete
+        /// </summary>
+        /// <param name="waitUntil"> <see cref="WaitUntil.Completed"/> if the method should wait to return until the long-running operation has completed on the service; <see cref="WaitUntil.Started"/> if it should return after starting the operation. For more information on long-running operations, please see <see href="https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/LongRunningOperations.md"> Azure.Core Long-Running Operation samples</see>. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public virtual async Task<ArmOperation<DeleteOperationResult>> DeleteAsync(WaitUntil waitUntil, CancellationToken cancellationToken = default)
```

Comment:
so the trailing /default is required for the delete methods but will break if you use it in the get methods?

Reply:
exactly